libyang/data.py

```diff
@@ -237,15 +237,18 @@ def __init__(self, context: "libyang.Context", cdata):
         self.cdata = cdata  # C type: "struct lyd_node *"
         self.free_func = None  # type: Callable[DNode]
 
-    def meta(self):
+    def meta(self, name=None):
         ret = {}
         item = self.cdata.meta
         while item != ffi.NULL:
-            ret[c2str(item.name)] = c2str(
+            val = c2str(
                 lib.lyd_value_get_canonical(
                     self.context.cdata, ffi.addressof(item.value)
                 )
             )
+            if name and c2str(item.name) == name:
+                return val
```

Comment:
I don't like the fact that this function may return a dict or a string value depending on the arguments.

We need another function that has a different signature.